File: 공소연/programmers/P12940.js
// 문제 링크: https://school.programmers.co.kr/learn/courses/30/lessons/12940
// 시간: 0.18ms
// 메모리: 33.4MB

function solution(n, m) {
    
    // 최대공약수
    let max = 0;
    for (let i = 1; i <= m; i++) {
        if (n % i === 0 && m % i === 0) max = Math.max(max, i);
    }
    
    // 최소공배수
    let min = (n * m) / max;
    
    return [max, min];
}

File: 오선민/programmers/P12940.java
/**
 * 문제 링크: https://school.programmers.co.kr/learn/courses/30/lessons/12940
 * 시간: 0.03 ms
 * 메모리: 89.4 MB
 * pr규칙: yymmdd 사이트명 문제번호 언어
 */

package programmers;

public class P12940 {
    public int[] solution(int n, int m) {
        return new int[]{gcd(n, m), lcm(n, m)};
    }

    private int gcd(int n, int m) {
        while (m != 0) {
            int temp = m;
            m = n % m;
            n = temp;
        }
        return n;
    }

    private int lcm(int n, int m) {
        return n*m / gcd(n, m);
    }
}
